### ISSUE TITLE
Allow currency groupings in inconsistent_digit_grouping

### ISSUE THREAD

**Issue by TopologicallySpeaking** [rust-lang/rust-clippy#6059]
I'm writing a program which stores some amount of cents as an unsigned integer. It's basically a fixed-point decimal. So, it makes sense to place an underscore before the last two digits, but clippy flips out over this. For example, the following does not pass the lint:

```rust
fn one_million_dollars() -> u64 {
    1_000_000_00
}
```

I propose that the behavior should be changed so that literals with no fractional part can have a final grouping of two digits. If everyone agrees, I'd be willing to implement the change myself.

**Comment by flip1995** [comment]
I think this is **really** specific to your use case. For such specific cases, where a lint doesn't make sense, because of some semantics that cannot be statically checked, `#[allow(..)]` exists.

You could also write this as 

```rust
fn one_million_dollars() -> u64 {
    1_000_000 * 100
}
```

which would make the intention clearer IMO, without losing performance since the expression gets cost evaluated anyway.

To represent other values than `00` for cents, you could use `x * 100 + y`, which again gets const evaluated.

Do you have other use cases in mind, where `xxx_yz` would make sense?

**Comment by TopologicallySpeaking** [comment]
No.
